File: src/main/java/com/exadel/training/service/SearchService.java
package com.exadel.training.service;

import com.exadel.training.model.Training;
import com.exadel.training.model.User;

import java.util.List;

/**
 * Created by HP on 29.07.2015.
 */
public interface SearchService {
    List<User> searchUsers(String searchWord);
    List<Training> searchTrainings(String searchWord);
}

File: src/main/webapp/page_common/searchService.js
angular.module("myApp.menuApp")
.factory("searchService",["$http","$q",function($http,$q){
  
  var searchApi = {};
  searchApi.searchTrainigs = function(query){
    return $http.post("/search_controller/search_trainings",query).success(function(data){
      console.log("Training Result for " + query +"\n" + JSON.stringify(data));
    });
  }
  searchApi.searchUsers = function(query){
    return $http.post("/search_controller/search_users",query).success(function(data){
      console.log("User Result for " + query +"\n" + JSON.stringify(data));
    });
  }
  
  searchApi.searchBoth = function(query){
    var users = searchApi.searchUsers(query);
    var tr = searchApi.searchTrainigs(query);
    
    return $q.all({users:users, trainings: tr});
  }
  
  return searchApi;
  
}])